### PR TITLE
Crreated Bug and feature request template

### PR DESCRIPTION
**Pull request by Jayakumar03**
Craeted the bug and feature request template for the repo.

### CODE REVIEW

**Comment by Hat52** [comment]
Why do we have both package-lock and yarn-lock file ?

**Review by ShaileshParmar11 — CHANGES_REQUESTED** [review]
Please remove package-lock and yarn.lock file, as you are not installing any new package

**Comment by Jayakumar03** [comment]
@ShaileshParmar11  I have tried adding the files in gitignore but the chages are occuring in git repo.

File: .gitignore [hunk 1]
```diff
@@ -18,6 +19,8 @@
 .env.test.local
 .env.production.local
 .env
+package-lock.json
+yarn.lock
```
Comment (Sachin-chaurasiya) [inline .gitignore:23]: Hello @ Jayakumar03, you miss understood, you don't have to put in the .gitignore file.

We recommend using `yarn` for development purposes so you should not be committing the `package-lock.json`.

Also, as you have not installed any new package so `yarn. lock` should not have any changes.

Please remove the `package-lock.json` and undo the changes to `yarn.lock` file.


```
 package-lock.json
yarn.lock
```

**Comment by Sachin-chaurasiya** [comment]
@Jayakumar03 , Can you link the corresponding issue with this PR?

**Comment by Jayakumar03** [comment]
@ShaileshParmar11 has asked to make the changes. Ideally it should not create an issue

**Comment by Jayakumar03** [comment]
@Sachin-chaurasiya @ShaileshParmar11 
I have removed package.lock.json and yarn file.

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya @ShaileshParmar11 I have removed package.lock.json and yarn file.

@Jayakumar03 , You have removed the `yarn.lock` file, [here](https://github.com/OpenSource-Journey/Your-Github-Contributions/pull/33#discussion_r1129119639) I have asked you to undo the changes of `yarn. lock` not to remove it, please check

**Comment by Sachin-chaurasiya** [comment]
> @ShaileshParmar11 has asked to make the changes. Ideally it should not create an issue

@Jayakumar03 , It will create an issue and we cannot have both `package-lock.json` and `yarn.lock` files.

File: .gitignore [hunk 2]
```diff
@@ -5,6 +5,7 @@
 /.pnp
 .pnp.js
 
+
```
Comment (Sachin-chaurasiya) [inline .gitignore:8]: @Jayakumar03 , Remove this extra space

File: yarn.lock [hunk 3]
```diff
@@ -10751,4 +10751,4 @@ yargs@^16.2.0:
 yocto-queue@^0.1.0:
   version "0.1.0"
   resolved "https://registry.yarnpkg.com/yocto-queue/-/yocto-queue-0.1.0.tgz#0294eb3dee05028d31ee1a5fa2c556a6aaf10a1b"
-  integrity sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==
+  integrity sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==
```
Comment (Sachin-chaurasiya) [inline yarn.lock:10754]: @Jayakumar03, Here we should have a newline at the end.

File: .github/ISSUE_TEMPLATE/BugReporting.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      value: |
+        Thanks for taking the time to fill out this bug report!
+  - type: input
+    id: contact
+    attributes:
+      label: Contact Details
+      description: How can we get in touch with you if we need more info?
+      placeholder: ex. email@example.com
+    validations:
+      required: false
```
Comment (Sachin-chaurasiya) [inline .github/ISSUE_TEMPLATE/BugReporting.yml:18]: @Jayakumar03 , We don't need the contact field, please remove it.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Left a few comments, thanks

**Comment by Jayakumar03** [comment]
@Sachin-chaurasiya  i have undone the changes to yarn.lock file and added pull request template also.

Sorry for the mistake. I am new to open source and this is the first time i have made an pull request.|
Going forward i will try to reduce these mistake.

File: yarn.lock (hunk 3 above)
Comment (Jayakumar03) [reply yarn.lock:10754]: shall i give two blank line or single blank line is enough ?

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya i have undone the changes to yarn.lock file and added pull request template also.
> 
> Sorry for the mistake. I am new to open source and this is the first time i have made an pull request.| Going forward i will try to reduce these mistake.

No worries, @Jayakumar03 , It's okay to make mistakes and learn from them, we at `Open Source Journey` believes in the process, not in the end result.

File: yarn.lock (hunk 3 above)
Comment (Sachin-chaurasiya) [reply yarn.lock]: [This](https://github.com/OpenSource-Journey/Your-Github-Contributions/pull/33/commits/68dc6dc4a7d49f1eb34232411a9dde576aeb1a2c) is good enough.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Left a few comments, thanks

**Comment by Jayakumar03** [comment]
Implemented suggestion in both the files

**Comment by Sachin-chaurasiya** [comment]
@Jayakumar03 , Pull request template should be inside the `.github` directory, please make that change as well.

File: .github/ISSUE_TEMPLATE/BugReporting.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+name: Bug Report
```
Comment (Sachin-chaurasiya) [inline .github/ISSUE_TEMPLATE/BugReporting.yml:1]: @Jayakumar03 , The File name should be `bug.yml`.

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+name: Feature Request
```
Comment (Sachin-chaurasiya) [inline .github/ISSUE_TEMPLATE/FeatureRequest.yml:1]: @Jayakumar03 , The File name should be `feature_request.yml`.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Left a few comments, thanks

File: .github/ISSUE_TEMPLATE/BugReporting.yml (hunk 5 above)
Comment (Jayakumar03) [reply .github/ISSUE_TEMPLATE/BugReporting.yml:1]: Done

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  - type: dropdown
+    id: browsers
+    attributes:
+      label: What browsers are you using
+      multiple: true
+      options:
+        - Firefox
+        - Chrome
+        - Safari
+        - Microsoft Edge
```
Comment (Sachin-chaurasiya) [inline .github/ISSUE_TEMPLATE/feature_request.yml:48]: @Jayakumar03 , I think we don't need the `browsers` dropdown for the feature request template. please remove it.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        Thanks for taking the time to fill out this Feature request report!
+
+  - type: textarea
+    id: what-happened
+    attributes:
+      label: Suggest an idea for this project ?
+      description: Also tell us, what did you expect to happen?
+      placeholder: Tell us what you see!
+    validations:
+      required: true
```
Comment (Sachin-chaurasiya) [inline .github/ISSUE_TEMPLATE/feature_request.yml:18]: Screenshots should be optional for both bug and feature request templates.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+  - type: textarea
+    id: screenshot
+    attributes:
+      label: Please provide an example screenshot
+      placeholder: screenshot
+    validations:
+      required: true
```
Comment (Sachin-chaurasiya) [inline .github/ISSUE_TEMPLATE/feature_request.yml:26]: Screenshots should be optional for both bug and feature request templates.

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
+## Pull Request Checklist
+
+Before submitting a pull request, please make sure that your changes meet the following criteria:
+
+- [ ] Your code is well-formatted and follows our coding conventions.
+- [ ] Your changes have been tested thoroughly and do not introduce any new bugs.
+- [ ] You have updated the documentation to reflect your changes.
+- [ ] You are Agreeing to the code of conduct
```
Comment (Sachin-chaurasiya) [inline .github/pull_request_template.md:8]:
```suggestion
- [ ]  Ensure that your code adheres to our coding conventions and is properly formatted.
- [ ] Test your changes extensively to ensure that they do not introduce any new bugs.
- [ ] Confirm that you agree to the code of conduct.
```

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 7 above)
Comment (Jayakumar03) [reply .github/ISSUE_TEMPLATE/feature_request.yml:48]: Done

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 9 above)
Comment (Jayakumar03) [reply .github/ISSUE_TEMPLATE/feature_request.yml:26]: done

File: .github/pull_request_template.md (hunk 10 above)
Comment (Jayakumar03) [reply .github/pull_request_template.md:8]: done

**Comment by Sachin-chaurasiya** [comment]
@Jayakumar03 , You have commented but I Can't see your changes, can you push the changes?

**Review by Sachin-chaurasiya — APPROVED** [review]
Great work @Jayakumar03  🙌🏻,
Thanks for the contributions ✨

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by Sachin-chaurasiya** [comment]
> @Jayakumar03 , Can you link the corresponding issue with this PR?

@Jayakumar03 , The last thing, let us know if you have any queries.

**Comment by Sachin-chaurasiya** [comment]
@ShaileshParmar11 , Please review.

**Comment by Jayakumar03** [comment]
Thanks @Sachin-chaurasiya

**Comment by Jayakumar03** [comment]
@Sachin-chaurasiya I have started learning typescript and I have already basic knowledge of react 

I want to start contributing to the repo code as well. can you suggest me any beginner level issue ?

**Comment by Jayakumar03** [comment]
Could also suggest me some guide for open source contributions ?

**Comment by Sachin-chaurasiya** [comment]
> Could also suggest me some guide for open source contributions ?

For guides, you can refer to this https://blog.sachinchaurasiya.dev/series/open-source

**Review by ShaileshParmar11 — APPROVED** [review]

**Comment by Sachin-chaurasiya** [comment]
@all-contributors please add @Jayakumar03  for Documentation
